tests/arch/arm/arm_irq_advanced_features/src/arm_irq_target_state.c

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright (c) 2020 Nordic Semiconductor ASA.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <ztest.h>
+#include <arch/cpu.h>
+#include <arch/arm/aarch32/cortex_m/cmsis.h>
+
+#if defined(CONFIG_ARM_SECURE_FIRMWARE) && \
+	defined(CONFIG_ARMV7_M_ARMV8_M_MAINLINE)
+
+extern irq_target_state_t irq_target_state_set(unsigned int irq,
+	irq_target_state_t target_state);
+extern int irq_target_state_is_secure(unsigned int irq);
+
+void test_arm_irq_target_state(void)
+{
+	/* Determine an NVIC IRQ line that is not currently in use. */
+	int i;
+
+	for (i = CONFIG_NUM_IRQS - 1; i >= 0; i--) {
+		if (NVIC_GetEnableIRQ(i) == 0) {
+			/*
+			 * Interrupts configured statically with IRQ_CONNECT(.)
+			 * are automatically enabled. NVIC_GetEnableIRQ()
```

Comment:
Make it clear that an interrupt is in use IFF it has been configured with IRQ_CONNECT()
```suggestion
			 * In use interrupts are automatically enabled by
			 * virtue of IRQ_CONNECT(.). NVIC_GetEnableIRQ()
```